server/service/service_packs.go

```diff
@@ -2,20 +2,47 @@ package service
 
 import (
 	"context"
+	"fmt"
 	"github.com/fleetdm/fleet/v4/server/authz"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 )
 
-func (svc *Service) ApplyPackSpecs(ctx context.Context, specs []*fleet.PackSpec) error {
+func (svc *Service) ApplyPackSpecs(ctx context.Context, specs []*fleet.PackSpec) ([]*fleet.PackSpec, error) {
 	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionWrite); err != nil {
-		return err
+		return nil, err
 	}
 
-	if err := svc.ds.ApplyPackSpecs(specs); err != nil {
-		return err
+	packs, err := svc.ds.ListPacks(fleet.PackListOptions{IncludeSystemPacks: true})
+	if err != nil {
+		return nil, err
 	}
 
-	return svc.ds.NewActivity(
+	namePacks := make(map[string]*fleet.Pack, len(packs))
+	for _, pack := range packs {
+		namePacks[pack.Name] = pack
+	}
+
+	var result []*fleet.PackSpec
+
+	// loop over incoming specs filtering out possible edits to Global or Team Packs
+	for _, spec := range specs {
+		// check to see if incoming spec is already in the list of packs
+		if p, ok := namePacks[spec.Name]; ok {
```

Comment:
This has a race condition, where you can create a pack named `team-1` before you create a team. I wonder if we should solve this by doing two things:

1. Make the unique constraint not only on name, but on name+type
2. Reflect that in here by using `"%s\0%s", spec.Name, spec.Type` or something like that as the key

Alternatively, we can also go into this with this known limitation. Not feeling super strongly about either, but it might cause issues.

Reply:
Yeah great point. This was getting a little messy because Specs don't have the concept of Type, so I was hoping we could have some discussion around how exactly we wanted to protect against modifications. I think option 1 makes the most sense to me and seems the least invasive in the code. It also enforces this at the DB level, which to your points earlier is probably the best place to keep these constraints.

Reply:
Well, 1 is part of it. If we allow more than one `team-1` this check might not work.

Reply:
FWIW I'm fine with leaving it as a known limitation if no one feels strongly.

Reply:
Given the inconsistency, could we make a ticket and/or comment the code for this limitation at least? It might never matter, but in 1 week I'll forget all about this conversation.

Reply:
Code comment sounds good to me.